### DISCUSSION TITLE
posts/how-to-use-appsettings/

### DISCUSSION THREAD

**Discussion by giscus** [BYJRK/byjrk.github.io#12] [Announcements]
# posts/how-to-use-appsettings/

appsettings.json 是一个相较于 App.config 更加灵活的配置文件，是 .NET Core 以来新增的一种配置方式，提供了更多的灵活性

https://blog.coldwind.top/posts/how-to-use-appsettings/

**Comment by c77777771** [comment]
在Debug和Release下IConfiguration的Sections属性有值。但是只有在Debug调用var configurationSection = _configuration.GetSection("DateOptions");能获取到值。在Release下获取不到值，且代码直接跳过了。

**Reply by c77777771** [reply]
WPF在.NET6环境下

**Reply by BYJRK** [reply]
你确定Release输出路径存在appsettings.json文件吗？

**Comment by show2723** [comment]
请问UP如何从appsettings.json获取多态对象
比如父类
~~~c#
class UserSettings
{
    public string? User { get; set; }
    public string? Password { get; set; }
}
~~~
子类
~~~c#
class SuperSettings: UserSettings
{
    public string? Super { get; set; }
}
~~~
如果采用
~~~c#
var userSettings = configuration.GetSection("UserSettings").Get<UserSettings>();
~~~
只能获得父类信息，而不能获取子类信息

**Reply by BYJRK** [reply]
为什么会有这种需求？这里你不能事先知道配置项具体对应什么子类吗？
